src/main/scala/org/bitcoins/core/crypto/TransactionSignatureSerializer.scala

```diff
@@ -148,13 +147,13 @@ trait TransactionSignatureSerializer extends RawBitcoinSerializerHelper with Bit
     if (inputIndex >= UInt32(spendingTransaction.inputs.size)) {
       logger.warn("Our inputIndex is out of the range of the inputs in the spending transaction")
       errorHash
-    } else if(hashType == SIGHASH_SINGLE && inputIndex >= UInt32(spendingTransaction.outputs.size)) {
+    } else if(HashType.isSIGHASH_SINGLE(hashType.num) && inputIndex >= UInt32(spendingTransaction.outputs.size)) {
```

Comment:
probably cleaner to call `hashType.isInstanceOf[SIGHASH_SINGLE]` instead of calling `isSIGHASH_SINGLE`